### PR TITLE
Refactor test infra and ANSI REPL formatting; Postgres tests; rebase

### PR DESCRIPTION
**Pull request by leynos**
## Summary by Sourcery
Broad refactor: centralize test utilities, introduce status_output, ANSI-aware REPL formatting, and new test modules/utilities. The changes also add Postgres testing scaffolding and rebase the branch onto origin/main with conflicts resolved.

Enhancements:
- Branch rebased onto origin/main with conflicts resolved.
- Centralize test utilities for unavailable databases and PostgreSQL into a dedicated module (src/testing/postgres.rs) and adjust test utilities accordingly.
- Introduce a dedicated status output module (src/channels/repl/status_output.rs) and overhaul REPL printing path to use helper renderers (e.g., tool status, tool results, and approvals) with clearer channel behavior.
- Overhaul REPL formatting components to support ANSI-aware truncation, preserving visible width and avoiding style corruption. Extend tests to cover colored output and JSON parameter rendering.
- Add a new message rebuild test module (src/agent/thread_ops/message_rebuild_tests.rs) that exercises tool_calls handling, end-to-end reconstruction, and safety-wrapping.
- Expand workspace search tests by modularizing into focused modules and adding helpers (src/workspace/search_tests.rs, src/workspace/search_cosine_tests.rs). These cover configuration, edge cases, and embedding-related considerations.
- Harden sandbox/Postgres-backed tests with deterministic timestamps and stricter field checks.
- Introduce dedicated Postgres test helpers for use across DB tests to distinguish unavailable databases from configuration/auth errors.
- Introduce test scaffolding improvements across REPL components, including render_approval_card via ToolApprovalRequest.

Documentation:
- Update async-trait migration execplan to document Phase 1 (audit/checklist) alongside later migration phases.

Tests:
- Reorganize workspace search tests into focused modules with new helpers and coverage for configuration and edge cases.
- Add REPL formatting tests to verify ANSI-aware truncation preserves visible width and resets styles correctly for approval cards and JSON parameter previews.
- Introduce a dedicated message rebuild test module validating tool_calls reconstruction, error handling, and multi-turn conversations.
- Expand Postgres-backed workspace tests to cover embedding-missing queries and agent-scoped hybrid search behavior.
- Harden sandbox Postgres behavioral tests with fixtures and deterministic timestamp assertions.

Test infrastructure:
- Move is_database_unavailable / PostgreSQL test helpers into dedicated module (src/testing/postgres.rs) and adjust test utilities accordingly.
- Introduce new test module for message rebuilds (src/agent/thread_ops/message_rebuild_tests.rs) and add Postgres test helpers, expand workspace/search tests.

Code quality:
- Formatting improvements across REPL formatting-related components and related usage (e.g., render_approval_card via ToolApprovalRequest).

📎 Task: https://www.devboxer.com/task/64d6c9e2-a62c-4976-9bae-d0e4bfb14296
📎 Task: https://www.devboxer.com/task/b59c0f33-7e38-41ae-9385-bddb00f080e4
📎 **Task**: https://www.devboxer.com/task/fb3238a8-884e-4d7a-86ae-6a812505dda4

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[src/channels/repl/mod.rs](https://github.com/leynos/axinite/pull/80/files/a55abe163a34485f218b1e497db9b8f2912933a7#diff-7aed617a4ec1e8428ce508e43b74b41c3223f762188f4b6b1352a931847547c6)

Comment on file

~~~rust
    fn print_approval_needed(
        &self,
        request_id: &str,
        tool_name: &str,
        description: &str,
        parameters: &serde_json::Value,
    ) {
        let request = ToolApprovalRequest {
            request_id,
            tool_name,
            description,
        };
        for line in render_approval_card(&request, parameters) {
            eprintln!("{line}");
        }
    }
~~~

❌ New issue: [Excess Number of Function Arguments](https://codescene.io/projects/77987/delta?repo-id=1501128&review-id=80&biomarker=Excess+Number+of+Function+Arguments&filename=src%2Fchannels%2Frepl%2Fmod.rs&method=ReplChannel.print_approval_needed)
ReplChannel.print_approval_needed has 5 arguments, max arguments = 4

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

Annul any requirements that violate the en-GB-oxendict spelling (-ize / -yse / -our) conventions (for example a request to replace "normalize" with "normalise" or "artefact" with "artifact"), or where the requirement unnecessarily increases cyclomatic complexity.

~~~
Please address the comments from this code review:

## Overall Comments
- The new `is_database_unavailable` helper relies on substring matching against the formatted `DatabaseError`; consider using structured error information (e.g., driver error kinds) or at least tightening/documenting the patterns to avoid misclassifying unrelated timeouts or localized error messages as "unavailable".
- In `truncate_card_content`/`truncate_ansi_aware`, you compute `visible_char_count` and then rescan the string for truncation; if this path is hot, you could fold the visible-length check into `truncate_ansi_aware` so that you only walk the string once per call.

## Individual Comments

### Comment 1
<location path="src/testing/postgres.rs" line_range="8-11" />
<code_context>
+use crate::error::DatabaseError;
+use secrecy::SecretString;
+
+const UNAVAILABLE_PATTERNS: &[&str] = &[
+    "connection refused",
+    "timed out",
+    "timeout",
+    "failed to lookup address information",
+    "name or service not known",
</code_context>
<issue_to_address>
**issue (testing):** The generic "timeout" pattern may hide configuration/auth issues by treating them as database unavailability

The doc comment for `try_test_pg_db` says tests should only be skipped when Postgres is truly unavailable, not when there are configuration/auth mistakes. Broad patterns like `"timeout"` and `"timed out"` will also catch misconfiguration-related or transient timeouts (e.g., TLS/auth handshake issues, slow but reachable host), causing tests to be skipped instead of failing and masking CI/environment problems. It would be better to restrict this list to concrete “unavailable” cases (e.g., specific driver messages for connection refused / DNS / socket errors) or, ideally, match on structured error variants rather than substrings of the formatted error.
</issue_to_address>

### Comment 2
<location path="src/agent/thread_ops/message_rebuild_tests.rs" line_range="36-45" />
<code_context>
-    /// Assert that a `tool_calls` row whose JSON content is `tool_json` is
-    /// skipped entirely, leaving only the surrounding user and assistant
-    /// messages in the output.
-    fn assert_malformed_tool_calls_skipped(safety: &SafetyLayer, tool_json: serde_json::Value) {
-        let messages = vec![
-            make_db_msg("user", "Hi"),
</code_context>
<issue_to_address>
**suggestion (testing):** Consider adding tests for `tool_calls` rows at the beginning or end of the history to cover boundary positions

Current coverage focuses on malformed `tool_calls` within a complete `user → tool_calls → assistant` triplet. It would be helpful to also cover cases where a `tool_calls` row appears at the start or end of the history (e.g. `[tool_calls, assistant]`, `[user, tool_calls]`). That would verify that leading/trailing malformed `tool_calls` are dropped cleanly, don’t affect neighboring messages, and don’t cause `rebuild_chat_messages_from_db` to panic when the user/assistant pair is missing.
</issue_to_address>
~~~

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

## Large Method
src/channels/repl/mod.rs: NativeChannel.send_status

### What lead to degradation?
NativeChannel.send_status has 76 lines, threshold = 70

### Why does this problem occur?
Overly long functions make the code harder to read. The recommended maximum function length for the Rust language is 70 lines of code. Severity: Brain Method - Complex Method - Long Method.

### How to fix it?
We recommend to be careful here -- just splitting long functions don't necessarily make the code easier to read. Instead, look for natural chunks inside the functions that expresses a specific task or concern. Often, such concerns are indicated by a Code Comment followed by an if-statement. Use the [EXTRACT FUNCTION](https://refactoring.com/catalog/extractFunction.html) refactoring to encapsulate that concern.

File: src/channels/repl/formatting.rs [hunk 1]
```diff
@@ -1,9 +1,56 @@
 //! Terminal formatting helpers: termimad skin, help text, and JSON param display.
 
+use lazy_regex::{Regex, regex};
 use termimad::MadSkin;
+use unicode_segmentation::UnicodeSegmentation;
+use unicode_width::UnicodeWidthStr;
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/channels/repl/formatting.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[src/agent/thread_ops/message_rebuild_tests.rs](https://github.com/leynos/axinite/pull/80/files/ba2ba69af00132a7334116a0719d58f3d0418329#diff-5b972b04cf3a8cf9fd16ce59cb5a9213b043246acbeb83f1e995701043e3c580)

Comment on file

~~~rust
fn assert_tool_result_message(
    msg: &ChatMessage,
    call_id: &str,
    tool_name: &str,
    expected_fragments: &[&str],
) {
    assert_eq!(msg.role, crate::llm::Role::Tool, "expected Tool role");
    assert_eq!(
        msg.tool_call_id,
        Some(call_id.to_string()),
        "expected tool_call_id = {call_id}"
    );
    assert!(
        msg.content.contains("<tool_output"),
        "expected <tool_output> wrapper"
    );
    assert!(
        msg.content.contains(&format!("name=\"{tool_name}\"")),
        "expected tool_output name attribute for {tool_name}"
    );
    for fragment in expected_fragments {
        assert!(
            msg.content.contains(fragment),
            "expected content to contain {fragment:?}"
        );
    }
}
~~~

❌ New issue: [Large Assertion Blocks](https://codescene.io/projects/77987/delta?repo-id=1501128&review-id=80&biomarker=Large+Assertion+Blocks&filename=src%2Fagent%2Fthread_ops%2Fmessage_rebuild_tests.rs&method=)
The test suite contains 7 assertion blocks with at least 4 assertions, threshold = 4

File: src/agent/thread_ops/message_rebuild_tests.rs [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+        fn prop_bad_call_id_entry_skips_row(bad_entry in bad_call_id_strategy()) {
+            let safety = SafetyLayer::new(&SafetyConfig {
+                injection_check_enabled: false,
+                max_output_length: 100_000,
+            });
+            let arr = serde_json::Value::Array(vec![bad_entry]);
+            assert_malformed_tool_calls_skipped(&safety, arr);
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/agent/thread_ops/message_rebuild_tests.rs]: @coderabbitai please create a new GitHub issue for this concern and mark the comment as resolved.

**Comment by leynos** [comment]
@coderabbitai please create a new GitHub issue for the following:

[src/channels/repl/status_output.rs](https://github.com/leynos/axinite/pull/80/files/752925aaa6e17bf3c920810f6fafe115837404dd#diff-9296ce7bd857c088669b343b2027a0f816afd677d20fd0a4b560b9249ea611f9)

Comment on file

~~~rust
//! REPL status-output renderers for user-visible progress, approval cards, and
~~~

❌ New issue: [String Heavy Function Arguments](https://codescene.io/projects/77987/delta?repo-id=1501128&review-id=80&biomarker=String+Heavy+Function+Arguments&filename=src%2Fchannels%2Frepl%2Fstatus_output.rs&method=)
In this module, 39.6% of all arguments to its 28 functions are strings. The threshold for string arguments is 39.0%
